src/jit/compiler.h

```diff
@@ -340,6 +340,40 @@ class  LclVarDsc
     }
 #endif // FEATURE_MULTIREG_ARGS
 
+    bool lvIsHfa() const
+    {
+#ifdef FEATURE_HFA
+        return _lvIsHfa;
+#else
+        return false;
+#endif
+    }
+
+    // returns 0 is this LclVar is not an HFA
+    // otherwise returns 1-4 indicating the number of register slots used by the HFA
+    unsigned lvHfaSlots() const
+    {
+#ifdef FEATURE_HFA
+        if (lvIsHfa())
+        {
+            assert(lvType==TYP_STRUCT);
+            if (lvHfaTypeIsFloat)
+            {
+                return lvExactSize / sizeof(float);
+            }
+            else
+            {
+                return lvExactSize / sizeof(double);
+            }
+        }
+        else
+#endif
+        {
+            return 0;
+        }
```

Comment:
I am not sure it is a good idea to define this routine to return 0 for non-hfa types.  I would suggest to assert that this method is called only on vars that are known to be a hfa

e.g.
```
assert(lvIsHfa());

#ifdef FEATURE_HFA
assert(lvType == TYP_STRUCT)
...
#else
  unreached();
#endif
```